classes/author/DAO.inc.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * @file classes/publication/DAO.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class publication
+ *
+ * @brief Read and write publications to the database.
+ */
+
+namespace APP\author;
+
+use APP\core\Application;
+use PKP\db\DAORegistry;
+use stdClass;
+use Illuminate\Support\Facades\DB;
+
+class DAO extends \PKP\author\DAO
+{
+    public $chapterAuthorRelationshipTable = 'submission_chapter_authors';
+
+    /**
+     * Add an author to a chapter.
+     */
+    public function insertChapterAuthor(int $authorId, int $chapterId, bool $isPrimary = false, int $sequence = 0)
+    {
+        DB::table($this->chapterAuthorRelationshipTable)->insert([
+            'author_id' => $authorId,
+            'chapter_id' => $chapterId,
+            'primary_contact' => $isPrimary,
+            'seq' => $sequence,
+        ]);
+    }
+
+    /**
+     * Remove an author from a chapter
+     */
+    public function deleteChapterAuthorById(int $authorId, int $chapterId)
+    {
+        DB::table($this->chapterAuthorRelationshipTable)
+            ->where('author_id', '=', (int) $authorId)
```

Comment:
The typehints already specify ints, so no need to add the casts.

Reply:
Fixed